test/unit/spec/VerifyWebauthn_spec.js

```diff
@@ -0,0 +1,271 @@
+        }
+        var beacon = new Beacon($sandbox);
+        return {
+          router: router,
+          form: forms,
+          beacon: beacon,
+          ac: authClient,
+          setNextResponse: setNextResponse,
+          successSpy: successSpy
+        };
+      });
+  }
+
+  var testAuthData = 'c29tZS1yYW5kb20tYXR0ZXN0YXRpb24tb2JqZWN0';
+  var testClientData = 'c29tZS1yYW5kb20tY2xpZW50LWRhdGE=';
+  var testSignature = 'YWJjZGVmYXNkZmV3YWZrbm1hc2xqZWY=';
+  var testCredentialId = 'vdCxImCygaKmXS3S_2WwgqF1LLZ4i_2MKYfAbrNByJOOmSyRD_STj6VfhLQsLdLrIdgvdP5EmO1n9Tuw5BawZw';
+  var testChallenge = 'kygOUtSWURMv_t_Gj71Y';
+
+  function mockWebauthn() {
+    navigator.credentials = {
+      get: function () {
+      }
+    };
+  }
+
+  function mockU2fSuccessVerify() {
+    window.u2f = { sign: function () {} };
+
+    spyOn(window.u2f, 'sign').and.callFake(function (appId, challenge, registeredKeys, callback) {
+      callback({
+        signatureData: testSignature,
+        clientData: testClientData
+      });
+    });
+  }
+
+  function mockWebauthnSignFailure() {
+    spyOn(navigator.credentials, 'get').and.callFake(function () {
+      var deferred = Q.defer();
+      deferred.reject({ message: 'something went wrong' });
+      return deferred.promise;
+    });
+  }
+
+  function mockWebauthnSignSuccess() {
+    spyOn(navigator.credentials, 'get').and.callFake(function () {
+      var deferred = Q.defer();
+      deferred.resolve({
+        response: {
+          signature: CryptoUtil.strToBin(testSignature),
+          clientDataJSON: CryptoUtil.strToBin(testClientData),
+          authenticatorData: CryptoUtil.strToBin(testAuthData)
+        }
+      });
+      return deferred.promise;
+    });
+  }
+
+  function mockWebauthnSignPending() {
+    spyOn(navigator.credentials, 'get').and.returnValue(Q.defer().promise);
+  }
+
+  function setupWebauthnFactor(options) {
+    options || (options = {});
+
+    spyOn(webauthn, 'isWebauthnOrU2fAvailable').and.returnValue(options.u2fSupported == true);
+    spyOn(webauthn, 'isNewApiAvailable').and.returnValue(options.webauthnSupported == true);
+
+    mockWebauthn();
+    if (options.signStatus === 'fail') {
+      mockWebauthnSignFailure();
+    } else if (options.signStatus === 'success') {
+      mockWebauthnSignSuccess();
+    } else {
+      mockWebauthnSignPending();
+    }
+
+    return setup();
+  }
+
+  function expectHasRightBeaconImage(test, desiredClassName) {
+    expect(test.beacon.isFactorBeacon()).toBe(true);
+    expect(test.beacon.hasClass(desiredClassName)).toBe(true);
+  }
+
+  function expectTitleToBe(test, desiredTitle) {
+    expect(test.form.titleText()).toBe(desiredTitle);
+  }
+
+  Expect.describe('Webauthn Factor', function () {
+    itp('shows the right beacon and title for webauthn', function () {
+      return setupWebauthnFactor({webauthnSupported: true, u2fSupported: true}).then(function (test) {
+        expectHasRightBeaconImage(test, 'mfa-u2f');
```

Comment:
should be `mfa-webauthn`

Reply:
Webauthn factor is reusing U2F images and texts for beacon.